ear-production-suite-plugins/plugins/object/src/object_frontend_connector.cpp

```diff
@@ -557,161 +560,144 @@ void ObjectsJuceFrontendConnector::trackPropertiesChanged(
+    return;
+  }
+  if(auto diffuseSlider = lockIfSame(diffuseSlider_, slider)) {
+    performSliderAction(p_->getDiffuse(), diffuseSlider, action);
+    return;
+  }
+  if(auto factorSlider = lockIfSame(factorSlider_, slider)) {
+    performSliderAction(p_->getFactor(), factorSlider, action);
+    return;
+  }
+  if(auto rangeSlider = lockIfSame(rangeSlider_, slider)) {
+    performSliderAction(p_->getRange(), rangeSlider, action);
+    return;
   }
 }
 
 void ObjectsJuceFrontendConnector::pannerValueChanged(
     ear::plugin::ui::PannerTopView* panner) {
-  if (!pannerTopView_.expired() && panner == pannerTopView_.lock().get()) {
+  if (auto pannerTopView = lockIfSame(pannerTopView_, panner)) {
     *(p_->getAzimuth()) = panner->getAzimuth();
     *(p_->getDistance()) = panner->getDistance();
   }
 }
 
 void ObjectsJuceFrontendConnector::pannerDragStarted(
     ear::plugin::ui::PannerTopView* panner) {
-  if (!pannerTopView_.expired() && panner == pannerTopView_.lock().get()) {
+  if (auto pannerTopView = lockIfSame(pannerTopView_, panner)) {
     p_->getAzimuth()->beginChangeGesture();
     p_->getDistance()->beginChangeGesture();
   }
 }
 
 void ObjectsJuceFrontendConnector::pannerDragEnded(
     ear::plugin::ui::PannerTopView* panner) {
-  if (!pannerTopView_.expired() && panner == pannerTopView_.lock().get()) {
+  if (auto pannerTopView = lockIfSame(pannerTopView_, panner)) {
     p_->getAzimuth()->endChangeGesture();
     p_->getDistance()->endChangeGesture();
   }
 }
 
 void ObjectsJuceFrontendConnector::pannerValueChanged(
     ear::plugin::ui::PannerSideView* panner) {
-  if (!pannerSideView_.expired() && panner == pannerSideView_.lock().get()) {
+  if (auto pannerSideView = lockIfSame(pannerSideView_, panner)) {
     *(p_->getElevation()) = panner->getElevation();
   }
 }
 
 void ObjectsJuceFrontendConnector::pannerDragStarted(
     ear::plugin::ui::PannerSideView* panner) {
-  if (!pannerSideView_.expired() && panner == pannerSideView_.lock().get()) {
+  if (auto pannerSideView = lockIfSame(pannerSideView_, panner)) {
     p_->getElevation()->beginChangeGesture();
   }
 }
 
 void ObjectsJuceFrontendConnector::pannerDragEnded(
     ear::plugin::ui::PannerSideView* panner) {
-  if (!pannerSideView_.expired() && panner == pannerSideView_.lock().get()) {
+  if (auto pannerSideView = lockIfSame(pannerSideView_, panner)) {
     p_->getElevation()->endChangeGesture();
   }
 }
 
 void ObjectsJuceFrontendConnector::comboBoxChanged(EarComboBox* comboBox) {
-  if (!routingComboBox_.expired() &&
-      comboBox == routingComboBox_.lock().get()) {
+  if (auto routingComboBox = lockIfSame(routingComboBox_, comboBox)) {
     *(p_->getRouting()) = comboBox->getSelectedEntryIndex();
   }
 }
 
 void ObjectsJuceFrontendConnector::buttonClicked(Button* button) {
-  if (!divergenceButton_.expired() &&
-      button == divergenceButton_.lock().get()) {
+  if (auto divergenceButton = lockIfSame(std::weak_ptr<Button>(divergenceButton_), button)) {
```

Comment:
Do we need to construct a weak_ptr here - isn't it already a weak_ptr?

Reply:
Thanks, I'm not sure how that crept in. I just like typing I guess.